File: java/src/main/java/com/galileucopilot/galileucopilot/entities/Vehicle.java
package com.galileucopilot.galileucopilot.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "table_vehicle")
public class Vehicle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String plate;
    private Integer capacity;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;

    @OneToMany(mappedBy = "vehicle")
    private List<Ride> rides;
}

File: experimento-copilot-back/Entities/Vehicle.cs
namespace experimento_copilot_back.Entities
{
    public class Vehicle
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string Plate { get; set; }
        public int Capacity { get; set; }

        public Guid OwnerId { get; set; }
        public User Owner { get; set; }
    }
}
